From b73a77622a634ea3e3bac323506ad51cb01907c1 Mon Sep 17 00:00:00 2001
From: Shivaprasad Hongal <shongal@codeaurora.org>
Date: Thu, 12 Jul 2018 16:24:45 -0700
Subject: [PATCH 5/8] fs_mgr: Wrapped key support for FBE

Add "ice_wrapped_key_supported" support to be read from
fstab.

CRs-Fixed: 2288316

Change-Id: I01f634e8d464059791d731d2af690bdfd8008107

# Conflicts:
#	fs_mgr/fs_mgr_fstab.cpp
---
 fs_mgr/fs_mgr_fstab.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/fs_mgr/fs_mgr_fstab.cpp b/fs_mgr/fs_mgr_fstab.cpp
index c9fb7aa57..8fc04c865 100644
--- a/fs_mgr/fs_mgr_fstab.cpp
+++ b/fs_mgr/fs_mgr_fstab.cpp
@@ -114,11 +114,13 @@ static struct flag_list fs_mgr_flags[] = {
 #define EM_ICE          2
 #define EM_AES_256_CTS  3
 #define EM_AES_256_HEH  4
+#define EM_ICE_WRAPPED_KEY_SUPPORTED 7
 
 static const struct flag_list file_contents_encryption_modes[] = {
     {"aes-256-xts", EM_AES_256_XTS},
     {"software", EM_AES_256_XTS}, /* alias for backwards compatibility */
-    {"ice", EM_ICE}, /* hardware-specific inline cryptographic engine */
+    {"ice", EM_ICE},              /* hardware-specific inline cryptographic engine */
+    {"ice_wrapped_key_supported", EM_ICE_WRAPPED_KEY_SUPPORTED}, /* ICE engine with wrapped key support */
     {0, 0},
 };
 
-- 
2.17.1

